packages/api/lambdas/sf-event-sqs-to-db-records/write-execution.js

```diff
@@ -1,69 +1,155 @@
-const log = require('@cumulus/common/log');
+const isNil = require('lodash/isNil');
+
 const {
   tableNames,
 } = require('@cumulus/db');
 const {
   getMessageExecutionArn,
+  getExecutionUrlFromArn,
   getMessageCumulusVersion,
+  getMessageWorkflowTasks,
+  getMessageWorkflowName,
+  getMessageWorkflowStartTime,
+  getMessageWorkflowStopTime,
+  getMessageExecutionOriginalPayload,
+  getMessageExecutionFinalPayload,
 } = require('@cumulus/message/Executions');
 const {
   getMetaStatus,
 } = require('@cumulus/message/workflows');
 
+const { parseException } = require('../../lib/utils');
 const Execution = require('../../models/executions');
 
 const {
   isPostRDSDeploymentExecution,
-  hasNoAsyncOpOrExists,
-  hasNoParentExecutionOrExists,
 } = require('./utils');
 
-const shouldWriteExecutionToRDS = async (
+const shouldWriteExecutionToRDS = ({
   cumulusMessage,
   collectionCumulusId,
-  knex
-) => {
+  asyncOperationCumulusId,
+  parentExecutionCumulusId,
+}) => {
   const isExecutionPostDeployment = isPostRDSDeploymentExecution(cumulusMessage);
   if (!isExecutionPostDeployment) return false;
 
-  try {
-    if (!collectionCumulusId) return false;
+  return !isNil(collectionCumulusId)
+    && !isNil(asyncOperationCumulusId)
+    && !isNil(parentExecutionCumulusId);
+};
 
-    const results = await Promise.all([
-      hasNoParentExecutionOrExists(cumulusMessage, knex),
-      hasNoAsyncOpOrExists(cumulusMessage, knex),
-    ]);
-    return results.every((result) => result === true);
-  } catch (error) {
-    log.error(error);
-    return false;
-  }
+const buildExecutionRecord = ({
```

Comment:
Nit: I'd feel a lot better if this was typescripted and used the Execution typings.     I also want a pony, so... totally admit this isn't scoped for this PR.  See comments above re: the organization/location of this lambda.